app/components/lead_providers/partnerships/challenged_banner.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module LeadProviders
+  module Partnerships
+    class ChallengedBanner < BaseComponent
+      def initialize(partnership:)
+        @partnership = partnership
+      end
+
+      def render?
+        partnership.challenged?
+      end
```

Comment:
Yup, this is a bit controversial part of view component and I am very much on the fence. On one hand, I very much agree with your argument - this just looks like a parent logic! On the other, the idea behind the component is to make it well-encapsulated so it should, theoretically, know when it can render...

But I don't really see much value in that code sitting in the component. I'll move it over.

Reply:
Actually, nah - it should not live on component even theoretically. If it is told to render it is to render or to die trying (exception).